pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/AggregationFunctionUtils.java

```diff
@@ -128,6 +79,29 @@ public static AggregationFunctionContext getAggregationFunctionContext(Aggregati
     return aggregationFunctions;
   }
 
+  /**
+   * Creates an {@link AggregationFunctionColumnPair} from the {@link AggregationFunction}. Returns {@code null} if
+   * the {@link AggregationFunction} cannot be represented as an {@link AggregationFunctionColumnPair} (e.g. has
+   * multiple arguments, argument is not column etc.).
+   */
+  @Nullable
+  public static AggregationFunctionColumnPair getAggregationFunctionColumnPair(
+      AggregationFunction aggregationFunction) {
+    AggregationFunctionType aggregationFunctionType = aggregationFunction.getType();
+    if (aggregationFunctionType == AggregationFunctionType.COUNT) {
+      return AggregationFunctionColumnPair.COUNT_STAR;
+    }
+    //noinspection unchecked
+    List<TransformExpressionTree> inputExpressions = aggregationFunction.getInputExpressions();
+    if (inputExpressions.size() == 1) {
+      TransformExpressionTree inputExpression = inputExpressions.get(0);
+      if (inputExpression.isColumn()) {
+        return new AggregationFunctionColumnPair(aggregationFunctionType, inputExpression.getValue());
+      }
+    }
+    return null;
```

Comment:
Are there other callers of this api, other than Star tree? If not, then may be better to throw exception?

Reply:
This is for star-tree only, but we use it to check whether the query can be solved by star-tree. We use `null` to notify the caller that the function cannot be solved by star-tree.